django_project/realtime/tasks/ash.py

```diff
@@ -36,42 +36,65 @@ def check_processing_task():
             task_status__iexact='SUCCESS'):
         task_id = ash.task_id
         result = AsyncResult(id=task_id, app=realtime_app)
-        ash.task_status = result.state
-        ash.save()
+        if result.state == 'SUCCESS':
+            Ash.objects.filter(id=ash.id).update(
+                task_status=result.state)
+            ash.refresh_from_db()
+            ash.save()
     # Checking analysis task
     for ash in Ash.objects.exclude(
             analysis_task_id__isnull=True).exclude(
             analysis_task_id__exact='').exclude(
             analysis_task_status__iexact='SUCCESS'):
         analysis_task_id = ash.analysis_task_id
         result = AsyncResult(id=analysis_task_id, app=headless_app)
-        ash.analysis_task_status = result.state
-        # Set the impact file path if success
-        if ash.analysis_task_status == 'SUCCESS':
-            ash.impact_file_path = result.result['output']['analysis_summary']
-        ash.save()
+
+        # Set impact file path if success
+        if result.state == 'SUCCESS':
+            task_state = 'FAILURE'
+            try:
+                ash.impact_file_path = result.result[
+                    'output']['analysis_summary']
+                task_state = 'SUCCESS'
+            except BaseException as e:
+                LOGGER.exception(e)
+
+            Ash.objects.filter(id=ash.id).update(
```

Comment:
Why don't we use something like this?
```python
ash.analysis_task_status = task_state
ash.save()
```

Reply:
When you hit .save() it will call the signals again which call generate_event_report.
If you do it using update like this, the signals won't be called. So it is useful to update task status, because in that case we don't want to trigger the signals.

Reply:
ok